yt_dlp/extractor/itv.py

```diff
@@ -113,11 +113,16 @@ def _call_api(self, video_id, playlist_url, headers, platform_tag, featureset, f
 
     def _get_subtitles(self, video_id, variants, ios_playlist_url, headers, *args, **kwargs):
         subtitles = {}
-        platform_tag_subs, featureset_subs = next(
-            ((platform_tag, featureset)
-             for platform_tag, featuresets in variants.items() for featureset in featuresets
-             if try_get(featureset, lambda x: x[2]) == 'outband-webvtt'),
-            (None, None))
+        platform_tag_subs = None
+        featureset_subs = None
+        # Prefer last matching featureset
+        # See: https://github.com/yt-dlp/yt-dlp/issues/986
+        for platform_tag, featuresets in variants.items():
+            for featureset in featuresets:
+                if 'outband-webvtt' in (featureset or []):
+                    platform_tag_subs = platform_tag
+                    featureset_subs = featureset
+                    break
         if platform_tag_subs or featureset_subs:
```

Comment:
Better to use `and` to make it clear both `platform_tag_subs` and `featureset_subs` are needed for API call

Reply:
Looks OK otherwise

Reply:
good catch, thanks